src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -635,6 +654,12 @@ AdaptiveExecutor(CitusScanState *scanState)
 
 	ExecuteSubPlans(distributedPlan);
 
+	bool hasDependentJobs = HasDependentJobs(job);
+	if (hasDependentJobs)
+	{
+		jobList = ExecuteDependentTasks(taskList, job);
```

Comment:
Returning the jobList here seems a little arbitrary. Maybe rebuild it in DoRepartitionCleanup? It's a very cheap operation.

Reply:
In order to rebuild it, we need the top level job, as far as I see we do not have it in execution. So I will put the top level job to the executor then to be able to reach to it from PG_CATCH where we only have execution.

Reply:
True, did not consider that. In that case the `jobIdList` is probably a simpler thing to keep than the top-level job.